app/controllers/concert_halls_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ConcertHallsController < ApplicationController
+  before_action :set_concert_hall, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @concert_halls = ConcertHall.all
+  end
+
+  def show
+  end
+
+  def new
+    @concert_hall = ConcertHall.new
+  end
+
+  def create
+    @concert_hall = ConcertHall.new(concert_hall_params)
+    if @concert_hall.save
+      redirect_to concert_hall_path(@concert_hall)
+    else
+      render :new
+    end
+  end
+
+  def edit
```

Comment:
You don't define edit action?

Reply:
edit and show only find with params[:id], defined before action

Reply:
edit and show only find with params[:id] wich defined before action